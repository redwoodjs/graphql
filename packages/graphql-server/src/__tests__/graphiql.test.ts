import { afterAll, beforeAll, vi, describe, expect, it } from 'vitest'

import { configureGraphiQLPlayground } from '../graphiql'

describe('configureGraphiQLPlayground', () => {
  describe('when not in development environment', () => {
    const curNodeEnv = process.env.NODE_ENV

    beforeAll(() => {
      process.env.NODE_ENV = 'not-development'
    })

    afterAll(() => {
      process.env.NODE_ENV = curNodeEnv
      expect(process.env.NODE_ENV).toBe('test')
    })

    it('should return false when no config is provided', () => {
      const result = configureGraphiQLPlayground({})

      expect(result).toBe(false)
    })

    it('should configure the GraphiQL Playground when allowGraphiQL is true', () => {
      const result = configureGraphiQLPlayground({
        allowGraphiQL: true,
        generateGraphiQLHeader: vi.fn(),
      })

      expect(result).not.toBe(false)
      expect(result).toMatchSnapshot()
    })

    it('should return false when allowGraphiQL is false', () => {
      const result = configureGraphiQLPlayground({
        allowGraphiQL: false,
        generateGraphiQLHeader: vi.fn(),
      })

      expect(result).toBe(false)
    })

    it('should return false when allowGraphiQL is not provided', () => {
      const result = configureGraphiQLPlayground({
        generateGraphiQLHeader: vi.fn(),
      })

      expect(result).toBe(false)
    })

    it('should return false when allowGraphiQL is undefined', () => {
      const result = configureGraphiQLPlayground({
        allowGraphiQL: undefined,
        generateGraphiQLHeader: vi.fn(),
      })

      expect(result).toBe(false)
    })

    it('should return false when allowGraphiQL is null', () => {
      const result = configureGraphiQLPlayground({
        // @ts-expect-error - We don't explicitly allow null, but we will cover
        // it in the tests anyway
        allowGraphiQL: null,
        generateGraphiQLHeader: vi.fn(),
      })

      expect(result).toBe(false)
    })
  })

  describe('when in development', () => {
    const curNodeEnv = process.env.NODE_ENV

    beforeAll(() => {
      process.env.NODE_ENV = 'development'
    })

    afterAll(() => {
      process.env.NODE_ENV = curNodeEnv
      expect(process.env.NODE_ENV).toBe('test')
    })

    it('should configure the GraphiQL Playground when no config is provided', () => {
      const result = configureGraphiQLPlayground({})

      expect(result).not.toBe(false)
      expect(result).toMatchSnapshot()
    })

    it('should configure the GraphiQL Playground when allowGraphiQL is true', () => {
      const result = configureGraphiQLPlayground({
        allowGraphiQL: true,
        generateGraphiQLHeader: vi.fn(),
      })

      expect(result).not.toBe(false)
      expect(result).toMatchSnapshot()
    })

    it('should configure the GraphiQL Playground when allowGraphiQL is false', () => {
      const result = configureGraphiQLPlayground({
        allowGraphiQL: false,
        generateGraphiQLHeader: vi.fn(),
      })

      expect(result).toBe(false)
    })

    it('should configure the GraphiQL Playground when allowGraphiQL is not provided', () => {
      const result = configureGraphiQLPlayground({
        generateGraphiQLHeader: vi.fn(),
      })

      expect(result).not.toBe(false)
      expect(result).toMatchSnapshot()
    })

    it('should configure the GraphiQL Playground when allowGraphiQL is undefined', () => {
      const result = configureGraphiQLPlayground({
        allowGraphiQL: undefined,
        generateGraphiQLHeader: vi.fn(),
      })

      expect(result).not.toBe(false)
      expect(result).toMatchSnapshot()
    })

    it('should configure the GraphiQL Playground when allowGraphiQL is null', () => {
      const result = configureGraphiQLPlayground({
        // @ts-expect-error - We don't explicitly allow null, but we will cover
        // it in the tests anyway
        allowGraphiQL: null,
        generateGraphiQLHeader: vi.fn(),
      })

      expect(result).not.toBe(false)
      expect(result).toMatchSnapshot()
    })
  })
})
