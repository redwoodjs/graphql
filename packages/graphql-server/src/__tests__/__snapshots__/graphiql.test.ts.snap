// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`configureGraphiQLPlayground > when in development > should configure the GraphiQL Playground when allowGraphiQL is not provided 1`] = `
{
  "defaultQuery": "query Redwood {
  redwood {
    version
  }
}",
  "headerEditorEnabled": true,
  "headers": undefined,
  "title": "Redwood GraphQL Playground",
}
`;

exports[`configureGraphiQLPlayground > when in development > should configure the GraphiQL Playground when allowGraphiQL is null 1`] = `
{
  "defaultQuery": "query Redwood {
  redwood {
    version
  }
}",
  "headerEditorEnabled": true,
  "headers": undefined,
  "title": "Redwood GraphQL Playground",
}
`;

exports[`configureGraphiQLPlayground > when in development > should configure the GraphiQL Playground when allowGraphiQL is true 1`] = `
{
  "defaultQuery": "query Redwood {
  redwood {
    version
  }
}",
  "headerEditorEnabled": true,
  "headers": undefined,
  "title": "Redwood GraphQL Playground",
}
`;

exports[`configureGraphiQLPlayground > when in development > should configure the GraphiQL Playground when allowGraphiQL is undefined 1`] = `
{
  "defaultQuery": "query Redwood {
  redwood {
    version
  }
}",
  "headerEditorEnabled": true,
  "headers": undefined,
  "title": "Redwood GraphQL Playground",
}
`;

exports[`configureGraphiQLPlayground > when in development > should configure the GraphiQL Playground when no config is provided 1`] = `
{
  "defaultQuery": "query Redwood {
  redwood {
    version
  }
}",
  "headerEditorEnabled": true,
  "headers": undefined,
  "title": "Redwood GraphQL Playground",
}
`;

exports[`configureGraphiQLPlayground > when not in development environment > should configure the GraphiQL Playground when allowGraphiQL is true 1`] = `
{
  "defaultQuery": "query Redwood {
  redwood {
    version
  }
}",
  "headerEditorEnabled": true,
  "headers": undefined,
  "title": "Redwood GraphQL Playground",
}
`;
